notebooks/samples/104 - Train, Test, Evaluate for Regression with Auto Imports Dataset.ipynb

```diff
@@ -0,0 +1,310 @@
+    "             (\"fuel-system\", StringType), (\"bore\", DoubleType), (\"stroke\", DoubleType), (\"compression-ratio\", DoubleType),\n",
+    "             (\"horsepower\", DoubleType), (\"peak-rpm\", DoubleType), (\"city-mpg\", LongType), (\"highway-mpg\", LongType),\n",
+    "             (\"price\",DoubleType))\n",
+    "\n",
+    "tableSchema = StructType([StructField(column[0], column[1](),True) for column in colSchema])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": true
+   },
+   "source": [
+    "Read the data from the AutomobilePriceRaw.csv file into a pandas dataframe. Specify possible reprsentations of missing values, and drop the 'num-of-doors' column as the data is read in. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "dataFile = \"AutomobilePriceRaw.csv\"\n",
+    "import os, urllib\n",
+    "if not os.path.isfile(dataFile):\n",
+    "    urllib.request.urlretrieve(\"https://mmlspark.azureedge.net/datasets/\"+dataFile, dataFile)\n",
+    "data = spark.createDataFrame(pd.read_csv(dataFile,\n",
+    "                                         na_values=['',' ','?'],\n",
+    "                                         usecols=lambda x: x not in ['num-of-doors']), tableSchema)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Summarize the data using `SummarizeData` and print the summary. Note that several columns have missing values (normalized-losses, bore, stroke, horsepower, peak-rpm, price)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "from mmlspark import SummarizeData\n",
+    "summary = SummarizeData().transform(data)\n",
+    "summary.toPandas()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now use the `CleanMissingData` API to replace the missing values with something more useful or meaningful. In this case, we will replace missing values in numeric columns with the median value for the column. Then, Summarize again and note the differences. Notice that all columns have 205 rows, and none of the rows contains missing values. Also, notice that the boundaries on the quartile bins have shifted slightly due to the missing value replacement."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "from mmlspark import CleanMissingData\n",
+    "cols = [\"normalized-losses\", \"stroke\", \"bore\", \"horsepower\", \"peak-rpm\", \"price\"]\n",
+    "cleanModel = CleanMissingData(cleaningMode=\"Median\", inputCols=cols, outputCols=cols).fit(data)\n",
```

Comment:
It'd be nice to show the Estimator property of CleanMissingData. Basically, split the data first, then fit the Estimator on training data and use it to transform train and test data. This could even be made into Pipeline together with TrainRegressor